redis_array_impl.c

```diff
@@ -466,7 +466,7 @@ ra_find_node(RedisArray *ra, const char *key, int key_len, int *out_pos TSRMLS_D
     if(!out) return NULL;
 
     if(ra->z_dist) {
-        if (!ra_call_distributor(ra, key, key_len, &pos TSRMLS_CC)) {
+        if (!ra_call_distributor(ra, out, out_len, &pos TSRMLS_CC)) {
```

Comment:
I don't actually know what the correct behavior would be here.  It seems logical to send the extracted part to the userland code, but then again perhaps not.  If we changed this and people were using their own extraction and distribution functions which ignored `{` and `}` it would break their code.

I don't know why they would be doing this, but I can bet within a day of changing the behavior I'll get bug reports :smiley: